### PR TITLE
Implement the confusable algorithm from Unicode more closely

### LINKED ISSUE #557: Higlight hack
There is hacker. He just write something like "**\* Noob has left the game" and everyone gots duplicate of this message as from server (yellow letters)
screenshot_2016-10-02_17-50-31

### LINKED ISSUE #575: Enhance nick similarity algorithm
Before applying the confusables mapping, we should do the Compatibility decomposition (NFKD, http://unicode.org/reports/tr15/) and stripping all combining characters.

That means in detail: Get the latest UnicodeData.txt (http://www.unicode.org/Public/9.0.0/ucd/UnicodeData.txt, are our Confusables.txt still up-to-date?), use it to replace every code point with the code points listed in column 5, strip all characters with a class beginning with "M" (Mark), specified in column 2.

### PR DESCRIPTION
**Pull request by heinrich5991**
See UTS#39 "Unicode Security Mechanisms":
http://www.unicode.org/reports/tr39/

This means that characters with accents or other things around them are
now considered the confusable with the base character.

Fixes #557. Fixes #575.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -9,6 +9,7 @@ SDL.dll
 SDL2.dll
 freetype.dll
 
+/confusables*
```
Comment (timakro) [inline .gitignore]: Which files does this exclude from git?

**Comment by timakro** [comment]
I don't know about this. It might lead to random collisions in names because the list is very long and also considers some chars as confusable which I personally don't find confusable at all. In particular dummy names might be affected (for example by dummy name `timakrơ` would collide with my player name). I didn't understand how the confusables are exactly formatted in `confusables_data.h`. I understood how the list is created and how the names are compared using that list in `confusables.c` which all looks fine. So if this works (I guess you tested it) the pr could be merged in terms of the code. If it is a good idea to merge this is another question.

**Comment by def-** [comment]
Performance is fine with this approach? I was worried about that a bit and that's one of the reasons I made a simpler implementation instead of the full approach.

**Comment by CytraL** [comment]
Only is used for nickname validations... i don't think that can have a big impact in the performance... perhaps i'm wrong.

¿what happens with not latin chars? ¿and why i can't use the nickname 'pingüino'? The problem with the confusable  characters for us depends of the font and the size... i think....

**Comment by heinrich5991** [comment]
Performance is fine. This is only done on name change.

**Comment by heinrich5991** [comment]
You can use the name pingüino if you want. This PR does nothing to stop you as far as I can tell.

**Comment by def-** [comment]
But it happens synchronously. If a 1ms delay could happen, that would be a huge problem.

**Comment by heinrich5991** [comment]
Thanks to deen, I now get 0.9µs per comparison.

**Comment by def-** [comment]
Worth nothing that my contribution consisted of asking "Performance is fine with this approach?" :D

**Comment by def-** [comment]
I'll just merge. Let's see what happens.

**Pull request merged**